frontend/src/pages/Fight.jsx

```diff
@@ -41,7 +47,18 @@ function Fight({ boss, hero }) {
       setBossLife(Math.max(bossLife - 1, 0));
     }
   };
-
+  const decrementStatIntelligence = () => {
+    setHeroIntelligence(Math.max(heroIntelligence - 1, 0));
+  };
+  const decrementStatSpeed = () => {
+    setHeroSpeed(Math.max(heroSpeed - 1, 0));
+  };
+  const decrementStatPower = () => {
+    setHeroPower(Math.max(heroPower - 1, 0));
+  };
+  const decrementStatStrength = () => {
+    setHeroStrength(Math.max(heroStrength - 1, 0));
+  };
```

Comment:
the decrementation can be done in useWeapon instead (once the states are reunited)